chain/ethereum/src/ethereum_adapter.rs

```diff
@@ -1123,71 +1118,29 @@ impl EthereumAdapterTrait for EthereumAdapter {
     fn block_hash_by_block_number(
         &self,
         logger: &Logger,
-        chain_store: Arc<dyn ChainStore>,
         block_number: BlockNumber,
-        block_is_final: bool,
     ) -> Box<dyn Future<Item = Option<H256>, Error = Error> + Send> {
         let web3 = self.web3.clone();
 
-        let mut hashes = match chain_store.block_hashes_by_block_number(block_number) {
-            Ok(hashes) => hashes,
-            Err(e) => return Box::new(future::result(Err(e))),
-        };
-        let num_hashes = hashes.len();
-        let logger1 = logger.clone();
-        let confirm_block_hash = move |hash: &Option<H256>| {
-            // If there was more than one hash, now that we know what the
-            // 'right' one is, get rid of all the others
-            if let Some(hash) = hash {
-                if block_is_final && num_hashes > 1 {
-                    chain_store
-                        .confirm_block_hash(block_number, hash)
```

Comment:
It's probably not a big deal in practice, but by never calling this, we'll hang on to forked blocks forever; duplicate blocks IIRC can lead to trouble resolving block numbers to block hashes. I don't think it's a big deal, since we don't make changes for nonfinal blocks anyway, but something to keep in mind.

Reply:
I kept the `fn confirm_block_hash` around, in case we can find a different spot for it. But I suppose an incoming firehose integration means we won't have to think to hard about this.